gnovm/pkg/gnolang/preprocess.go

```diff
@@ -457,6 +457,9 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 			case *AssignStmt:
 				checkValDefineMismatch(n)
 
+				for _, rx := range n.Rhs {
+					checkExprIsAssignable(rx)
+				}
```

Comment:
Should we not try to do this on leave, by evaluating what's the static type of the expression?

Reply:
Are you suggesting that we shift the check from `trans_enter` to `trans_leave`? We could implement it by adding a check for the assignment like this:

```go
for _, n := range n.Rhs {
	evalStaticType(store, last, n)
}
```

Reply:
Yeah, just so we evaluate it after we rewrite some of the expressions, I think it's a better place.

Reply:
yes it is better ^^
done here: https://github.com/gnolang/gno/pull/2896/commits/0646ab109fe22f6ed742fd791ff348bad18942ef